provenance/repos.py

```diff
@@ -510,12 +573,12 @@ def put(self, artifact_record, read_through=False):
                                s.serializer(artifact_record))
 
             inputs_json = _inputs_json(artifact_record.inputs)
-
-            db_artifact = db.Artifact(artifact_record, inputs_json)
+            run = self._current_run
+            db_artifact = db.Artifact(artifact_record, inputs_json, run)
```

Comment:
This will currently clobber existing runs when `read_through` is true. We need to create a new run from the existing `run_info` of the record if it exists.

If the current_run_info is actually put onto the artifact_record, then we wouldn't need to worry about checking read_through and could treat them all the same.

If we use UUID for the run ids, then they will be consistent across databases.